### DISCUSSION TITLE
Basic usage silently fails

### DISCUSSION THREAD

**Discussion by raphaelbastide** [djipco/webmidi#342] [General]
With Firefox Nightly (v111), I manage to see midi messages from my hardware without webmidi.js. But because it’s seams to be way easier to use the library to intercept them, I try to list inputs and outputs just like the basic page recommands.

```
WebMidi
  .enable()
  .then(onEnabled)
  .catch(err => alert(err));

function onEnabled() {
  console.log('test');
  
  // Inputs
  WebMidi.inputs.forEach(input => console.log(input.manufacturer, input.name));
  
  // Outputs
  WebMidi.outputs.forEach(output => console.log(output.manufacturer, output.name));

}
```
The result is: nothing really happens, except a security notice for midi access I accept. Then nothing is logged. If I `console.log(WebMidi);` I can see the object as expected (with inputs and outputs), but it seems the `onEnabled()` function is never called, and no error is caught. What do I miss?

**Comment by schoko11** [comment]
Hi, i think i came across something similar, where it seems to be some kind of problem with android(13) and firefox (109)...

did you use https?

**Reply by raphaelbastide** [reply]
No, I didn’t use https, I don’t know how to do so with localhost.

**Reply by schoko11** [reply]
localhost is fine and considered as secure origin

if you need to run local js in the browser you can start e.g. chrome with 
--allow-file-access-from-files

(but using a webserver seems to be a better option)
not sure if that helps you out

**Reply by raphaelbastide** [reply]
Actually I am a bit lost, even with Chromium I can’t make the library work with the code above. I am on Linux. Is it possible the problem comes from that?

**Reply by schoko11** [reply]
Hmm, well how do you setup your project, which platform etc. ,  "normally" you would use something like vscode + "live server"

**Reply by raphaelbastide** [reply]
Yes vscode + apache server. I must say that everything works well without the library. I can see midi hardware and receive midi events.

**Reply by djipco** [reply]
> Actually I am a bit lost, even with Chromium I can’t make the library work with the code above. I am on Linux. Is it possible the problem comes from that?

I just tried your code with Firefox v109 and it works for me. I'm on MacOS but it should work fine on Linux too. Which flavour are you using (IIFE or ESM) ?

**Reply by raphaelbastide** [reply]
Right now I use a CDN, IIFE.

**Reply by raphaelbastide** [reply]
![image](https://user-images.githubusercontent.com/193122/218771758-67bd68d0-1d11-4c75-8788-c2da811c531d.png)

**Reply by raphaelbastide** [reply]
The screenshot above shows I can log the Webmidi object but `test` never get logged.

**Reply by djipco** [reply]
Does it work with the regular Firefox (not the nightly) ?

**Reply by raphaelbastide** [reply]
No. Here is a screenshot with FF v.109.0.1. (Sorry for the screenshots, but I think that’s the best way to show you exactly what happens)
![image](https://user-images.githubusercontent.com/193122/221163029-67e98f76-9391-4bd1-8537-d2c17e86166e.png)

**Reply by raphaelbastide** [reply]
This issue was related to a Firefox installed on Linux using Snap. Snap installations mess with lots of system things, not an expert, but this should be known. If you encounter this problem, try installing FF / FF nightly with your distro’s package manager.
